cypress/integration/CTS/SearchResultsPageTests/index.js

```diff
@@ -0,0 +1,169 @@
+
+When('user clicks on chat button', () => {
+    cy.get('#chat-button').click();
+});
+
+And(`all result item's checkboxes are not checked`, () => {
+    cy.get('.results-list .cts-checkbox .cts-checkbox__label').each(($el) => {
+        const win = $el[0].ownerDocument.defaultView;
+        // use getComputedStyle to read the pseudo selector
+        const before = win.getComputedStyle($el[0], ':before');
+        // read the value of the `content` CSS property
+        const contentValue = before['background-image'];
+        // the returned value will have double quotes around it, but this is correct
+        expect(contentValue).to.eq('none');
+    });
+});
+
+And('{string} checkbox is displayed', (selectAllCheckbox) => {
+    cy.get(`label[for='select-all-cbx-top']`).should('be.visible').and('have.text', selectAllCheckbox);
+});
+
+When('user selects {string} checkbox', (selectAllCheckbox) => {
+    cy.get(`label[for='select-all-cbx-top']`).contains(selectAllCheckbox).click();
+});
+
+Then(`all result item's checkboxes are checked`, () => {
+    cy.get('.results-list .cts-checkbox .cts-checkbox__label').each(($el) => {
+        const win = $el[0].ownerDocument.defaultView;
+        // use getComputedStyle to read the pseudo selector
+        const before = win.getComputedStyle($el[0], '::before');
+        // read the value of the `content` CSS property
+        const contentValue = before['background-image'];
+        // the returned value will have double quotes around it, but this is correct
+        expect(contentValue).not.to.eq('none');
+    });
+});
+
+When(`user selects {int} result's checkbox`, (num) => {
+    cy.get(`label.cts-checkbox__label`).eq(num - 1).click();
+});
+
+Then('{string} checkbox is not checked', (selectAllCheckbox) => {
+    cy.get('label').contains(selectAllCheckbox).parent().find('input').should('not.be.checked');
+});
+
+And(`{int} result's checkbox is not checked`, (num) => {
+    cy.get('label.cts-checkbox__label').eq(num - 1).parent().find('input').should('not.be.checked');
+});
+
+Then('{string} checkbox is checked', (selectAllCheckbox) => {
+    cy.get('label').contains(selectAllCheckbox).parent().find('input').should('be.checked');
+});
+
+And(`{int} result's checkbox is checked`, (num) => {
+    And(`{int} result's checkbox is not checked`, (num) => {
+        cy.get('label').eq(num - 1).parent().find('input').should('be.checked');
+    });
+});
+
+And('user clicks on {string} button at {string} position', (printButton, printButtonPosition) => {
+    cy.get(`div.results-page__control.--${printButtonPosition}`).find('button.results-page__print-button').contains(printButton).click();
+});
+
+Then('user is redirected to {string} with generated {string}', (redirectedPath, printID) => {
+    cy.location('pathname').should('eq', redirectedPath);
+    cy.location('search').should('include', `?${printID}=`);
+});
+
+And('print page title is {string}', (title) => {
+    cy.get('h1').should('have.text', title);
```

Comment:
this should not be needed too - this should work with the generic step
page title is {string}

Reply:
Done